src/operator/controllers/kafkaacls/intentsAdmin.go

```diff
@@ -29,11 +29,19 @@ var (
 	namespaceRE   = regexp.MustCompile(`\$Namespace`)
 )
 
-type KafkaIntentsAdmin struct {
-	kafkaServer            otterizev1alpha1.KafkaServerConfig
-	kafkaAdminClient       sarama.ClusterAdmin
-	userNameMapping        string
-	enableKafkaACLCreation bool
+type KafkaIntentsAdmin interface {
+	ApplyServerTopicsConf(topicsConf []otterizev1alpha1.TopicConfig) error
+	ApplyClientIntents(clientName string, clientNamespace string, intents []otterizev1alpha1.Intent) error
+	RemoveClientIntents(clientName string, clientNamespace string) error
+	RemoveAllIntents() error
+	Close()
+}
+
+type KafkaIntentsAdminImpl struct {
+	KafkaServer            otterizev1alpha1.KafkaServerConfig
```

Comment:
I looked but couldn't find a reason for all the fields to be public?
They're also implementation details, so they should be private and internal to the KafkaIntentsAdminImpl

Reply:
It was necessary to create the object from the tests, I add `GetMockIntentsAdminFactory` function who create it inside the module to avoid making the fields public